public/app/features/plugins/loader/cache.test.ts

```diff
@@ -1,54 +1,64 @@
-import * as pluginSettings from '../pluginSettings';
+import { registerPluginInCache, invalidatePluginInCache, resolveWithCache, getPluginFromCache } from './cache';
 
-import { invalidatePluginInCache, resolveWithCache, registerPluginInCache } from './cache';
+jest.mock('./constants', () => ({
+  CACHE_INITIALISED_AT: 123456,
+}));
 
-describe('Plugin Cache', () => {
-  const now = 12345;
+describe('Cache Functions', () => {
+  describe('registerPluginInCache', () => {
+    it('should register a plugin in the cache', () => {
+      const plugin = { pluginId: 'plugin1', version: '1.0.0', isAngular: false };
+      registerPluginInCache(plugin);
+      expect(getPluginFromCache('plugin1')).toEqual(plugin);
```

Comment:
**Nit:** although this test is fine, I think `getPluginFromCache()` will not always give back the same result as what you registered (it's gonna be a different object, in certain cases with different fields due to `encodeURI(version)`).